src/amperity/gocd/secret/vault/plugin.clj

```diff
@@ -237,6 +238,8 @@
 (defmethod handle-request "go.cd.secrets.secrets-lookup"
   [client _ data]
   (try
+    (when-not @client
+      (authenticate-client-from-inputs! client (:configuration data)))
```

Comment:
Oh ok so the configuration for the plugin is just always sent during a request? Nice find!

Reply:
During a secret lookup specifically! I found that out by just logging all the requests like you suggested so thank you :)